packages/config-array/src/config-array.js

```diff
@@ -968,7 +966,7 @@ export class ConfigArray extends Array {
 	 * @param {Object} config The config to finalize.
 	 * @returns {Object} The finalized config.
 	 */
-	[ConfigArraySymbol.finalizeConfig](config) {
+	[/** @type {never} */ (ConfigArraySymbol.finalizeConfig)](config) {
```

Comment:
Can you add a comment somewhere in this file that explains why `never` is used? I know you explained in the PR description, but I can imagine revisiting this file at some time in the future and forgetting why this was done, so it would be good to have the explanation somewhere nearby.